crates/metaslang/bindings/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+pub mod builder;
+
+use std::collections::{BTreeSet, HashMap};
+use std::fmt;
+use std::fmt::Debug;
+use std::iter::once;
+
+use metaslang_cst::cursor::Cursor;
+use metaslang_cst::KindTypes;
+use metaslang_graph_builder::ast::File;
+use metaslang_graph_builder::functions::Functions;
+use semver::Version;
+use stack_graphs::graph::StackGraph;
+use stack_graphs::partial::PartialPaths;
+use stack_graphs::stitching::{ForwardPartialPathStitcher, GraphEdgeCandidates, StitcherConfig};
+use thiserror::Error;
+
+type Builder<'a, KT> = builder::Builder<'a, KT>;
+type GraphHandle = stack_graphs::arena::Handle<stack_graphs::graph::Node>;
+
+#[derive(Error, Debug)]
+pub enum BindingsError {
+    #[error(transparent)]
+    BuildError(#[from] builder::BuildError),
```

Comment:
IIUC, this is an internal error (happens because of a bug in our code), and not an actionable user error. Our tests should catch it if we have a bug in the `msgb` file.
I wonder if we should use `unreachable!()` in that case? it is not something we should expose to the user.

Reply:
That's a good point. I'll remove it for now and use `unreachable!()` or some other panicking function.

Reply:
I think we will need to reintroduce this in some form later in the development though. I was thinking of ways the bindings construction can fail, and one obvious case is a failure to resolve an import. That is something the user will control.